cEP-0016.md

```diff
@@ -0,0 +1,180 @@
+# Git Commit Content Inspection
+
+| Metadata |                                                   |
+| -------- | ------------------------------------------------- |
+| cEP      | 16                                                |
+| Version  | 1.0                                               |
+| Title    | Git commit content inspection                     |
+| Authors  | Kriti Rohillla <mailto:kritirohilla567@gmail.com> |
+| Status   | Implementation due                                |
+| Type     | Process                                           |
+
+## Abstract
+
+This document describes coala process rules which will be enforced by the
+GitCommitBear. Some new changes will be introduced in the implementation of
+GitCommitBear based on which the commits made to coala will be detected and
```

Comment:
Is this done in `GitCommitBear`, or `_CommitBear`?

Reply:
GitCommitBear

Reply:
I wish to do my project for git commits only. But IMO the changes should be done in `_CommitBear` using condition `if vcs_name == 'git'` because all the commit inspection is done in this bear only and in future, when someone will work on these changes for Hg commits, some of the code changes will be easily reusable. @jayvdb @damngamerz suggestions ?

Reply:
It will be easier to only worry about git and not hg also. Then the git specific logic needs to be in GitCommitBear  . We dont want git logic in the base class . That is bad OOP.
Some redesign is needed.
I strongly recommend that we look at making these three bears to be meta bears, like URLBear, so new custom bears can be created to do specific checks using git metadata.

Reply:
I think only one meta bear will be required. For commits that skip ci build, only filename of modified files should be sufficient and those should not be in `.coafile` or among test files. So this has little to do with the changed contents of files in the commit. Plus it will have a separate bear anyway https://github.com/coala/coala-bears/issues/1976 .
Analyzing changed files in the commit itself is major requirement for revert commits and other additional tasks in phase 2. I don't think we require it for merge commits either. So only *one* meta bear that takes the commit hash sent by `CommitBear` and returns list of modified files and no. of additions, deletions in each should do.

Reply:
A meta bear that provides lists of modified, added & deleted files is indeed one of the meta bears your project should create.

The other two meta bears would be merge and revert bears.  They would provide other bears `Result` objects which contain extra information.  e.g. a merge meta bear would provide multiple heads, and a revert meta bear would provide the reverted commit and the reverting commit. 

You will create only one bear which processes merges, and one bear which processes reverts.

But someone else might want different logic, and they could create their own bears which are activated only when there is a merge or a revert, re-using the meta-bear logic to detect these scenarios.